### ISSUE TITLE
508-defect-1 ❗ Launchblocker [SCREENREADER, COGNITION]: Button label is not comprehensible to screen reader users

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#23963]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh and Trevor

## User Story or Problem Statement

As a screen reader user, I want to know what content I am copying a link for.

## Details

The accessible label is being created by referencing the `id` of the accordion using `aria-label`. Because the dashes from those `id`s are carried over into the aria-label, JAWS will announce the button's label in an incomprehensible manner e.g. 

>  "Copy can dash I dash get dash a dash covid-19 dash vaccine dash a thirty six thousand nine hundred and ten sharable link, button."

NVDA and Safari won't announce the dashes, but the label is cut off at a certain character count before transitioning into a unique id-number (e.g. 36910) which is confusing.

## Acceptance Criteria

- [ ] Dashes are removed from the `aria-label` and are not announced
- [ ] Number (e.g. 36910) is removed from the `aria-label`
- [ ] Label is not cut off

## Environment

`https://staging.va.gov/coronavirus-veteran-frequently-asked-questions/`

## Steps to Recreate

Turn on JAWS, tab to the copy link button, confirm all dashes are announced.

## Recommended Solution

Beyond the stated acceptance criteria, it may be worth exploring from a content perspective if the "Copy link" portion of the label should be announced completely before the heading instead of being split apart. For example, instead of:

"Copy Can I get a COVID-19 vaccine at VA?, sharable link, button"

We may want to consider wording it as the following to avoid any confusion:

"Copy a share link for Can I get a COVID-19 vaccine at VA?, button"

@peggygannon 

## Screenshots or Trace Logs

<img width="868" alt="Screen Shot 2021-04-28 at 1 53 14 PM" src="https://user-images.githubusercontent.com/14154792/116460049-949cc580-a834-11eb-92a4-3586a6369c58.png">

**Comment by peggygannon** [comment]
Hi, @joshkimux to clarify, you're not recommending a change to any content on the page? This recommendation addresses how  how a screen reader would announce  content/sharable link or button?

**Comment by peggygannon** [comment]
Hi, @joshkimux Yes, I would agree. This recommendation is more plain language and understandable to users: 

"Copy a share link for Can I get a COVID-19 vaccine at VA?, button"

**Comment by SKasimow** [comment]
@youngfreezyVA Hi - can you take a look at this and lmk points and timing to amend as suggested ^. Thx

cc @jenniferlee-dsva

**Comment by youngfreezyVA** [comment]
this label is generated from the dataEntityID (dynamic)... so we would need to change the label to something static and need to update the original requirements to make this chnage
if we update the dynamically generated label there is the chance for regression issues, so i'd suggest a static label instead of this one:

`aria-label={`Copy ${dataEntityId} sharable link`}`

**Comment by SKasimow** [comment]
@youngfreezyVA  by changing the label "to something static" would the screen reader still read the text aloud like this (without  the dashes):

"Copy Can I get a COVID-19 vaccine at VA?, sharable link, button"

**Comment by youngfreezyVA** [comment]
no, b/c that content is dynamic. if we wanted something dynamic **_and_** legible to the screenreader, I could pass down the h3 content instead of the dataEntityID

**Comment by jenniferlee-dsva** [comment]
@youngfreezyVA - I like your idea for using the h3 text for the screenreader text. 

Per other comments above, let's make sure that that h3 text comes at the end of the announcement.
 
Per Peggy, like this: "Copy a share link for `Can I get a COVID-19 vaccine at VA` button"   The dynamic part coming from the `h3`.

**Comment by SKasimow** [comment]
Great - if you can change to Peggy/Josh's recommended phrasing as well, all the better.

**Comment by jenniferlee-dsva** [comment]
Made one small edit to Peggy's recommended text [here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/23963#issuecomment-840044809) -- please exclude the word "share." It is not a share link. It's just a link. We're just copying a link.  I don't know how this idea got started that this is some sort of special type of link with sharing behavior. :-(  

It's just a link, so let's not suggest otherwise to users, screen reader-assisted and otherwise.

**Comment by jenniferlee-dsva** [comment]
Sorry for the messy thought process. Now that I'm looking at the whole context of this - it feels weird to say copy a link and then button. Are we really trying to say that this is a button that lets you copy a link?

I suggest: 

Button to copy a link to "[insert h3 text]."

@joshkimux - I think this is still (maybe slightly more?) human friendly for screen reader text. Thoughts?

**Comment by joshkimux** [comment]
@jenniferlee-dsva We actually can't control where/when "button" will be announced as that's handled by the screen reader. Everything else we can control 👍

**Comment by SKasimow** [comment]
@jenniferlee-dsva @joshkimux @youngfreezyVA 

Question: does the screen reader always place "button" announcement last or does it vary?

And confirming final text is:

"Copy a link for [insert h3 text]"

**Comment by joshkimux** [comment]
@SKasimow It will usually be announced last. There may be screen readers that announce it first, though.

**Comment by jenniferlee-dsva** [comment]
> And confirming final text is:
> 
> "Copy a link for [insert h3 text]"

Yes please and thank you @SKasimow @youngfreezyVA @joshkimux

**Comment by joshkimux** [comment]
🥇 @youngfreezyVA 
<img width="442" alt="Screen Shot 2021-05-14 at 3 19 44 PM" src="https://user-images.githubusercontent.com/14154792/118318752-d0d45500-b4c7-11eb-8d1a-f7f7245b17dd.png">
